src/Cmd/Add.cc

```diff
@@ -124,8 +127,21 @@ addDependencyToManifest(
     }
   }
 
-  auto data = toml::parse(getManifestPath());
-  auto& deps = toml::find<toml::table>(data, "dependencies");
+  // Keep the order of the tables.
+  auto data = toml::parse<toml::ordered_type_config>(getManifestPath());
+
+  // Ensure the "dependencies" key exists and is a table for correct
+  // serialization.
+  if (data["dependencies"].is_empty()) {
+    data["dependencies"] = toml::table{};
+  }
+  if (!data["dependencies"].is_table()) {
+    logger::error(
+        "The `dependencies` key in the poac.toml is not the correct type"
```

Comment:
This error message is ambiguous. We should show what it should be:
```suggestion
        "The `dependencies` key in the poac.toml should be a table type"
```

Also, use [`make_error_info`](https://github.com/ToruNiina/toml11?tab=readme-ov-file#error-messages) to provide a clearer error message.